From ecbd275386e5608f0f90d3536506d7f50026ffe6 Mon Sep 17 00:00:00 2001
From: "Nathaniel J. Smith" <njs@pobox.com>
Date: Thu, 1 Aug 2019 21:23:51 -0700
Subject: [PATCH] Support openssl <1.1.1

Upstream-Status: Inappropriate
[https://github.com/pyca/pyopenssl/pull/861/commits/ecbd275386e5608f0f90d3536506d7f50026ffe6
was accepted in 19.1.0 but later removed since the upstream dropped
support of OpenSSL <1.1.1]
Signed-off-by: Stefan Strogin <steils@gentoo.org>
---
 src/OpenSSL/SSL.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/OpenSSL/SSL.py b/src/OpenSSL/SSL.py
index 6f785fc4..55211511 100644
--- a/src/OpenSSL/SSL.py
+++ b/src/OpenSSL/SSL.py
@@ -140,7 +140,10 @@ class _buffer(object):
 OP_NO_TLSv1 = _lib.SSL_OP_NO_TLSv1
 OP_NO_TLSv1_1 = _lib.SSL_OP_NO_TLSv1_1
 OP_NO_TLSv1_2 = _lib.SSL_OP_NO_TLSv1_2
-OP_NO_TLSv1_3 = _lib.SSL_OP_NO_TLSv1_3
+try:
+    OP_NO_TLSv1_3 = _lib.SSL_OP_NO_TLSv1_3
+except AttributeError:
+    pass
 
 MODE_RELEASE_BUFFERS = _lib.SSL_MODE_RELEASE_BUFFERS
 
